lms/djangoapps/course_home_api/course_metadata/v1/views.py

```diff
@@ -65,8 +66,14 @@ def get(self, request, *args, **kwargs):
         )
 
         course = course_detail(request, request.user.username, course_key)
-        user_is_enrolled = CourseEnrollment.is_enrolled(request.user, course_key_string)
-
+        enrollment = CourseEnrollment.get_enrollment(request.user, course_key_string, select_related=['celebration'])
+        user_is_enrolled = enrollment is not None
+        should_celebrate_streak = False
+        if not is_masquerading_as_specific_student(request.user, course_key):
+            should_celebrate_streak = UserCelebration.perform_streak_updates(request.user, course_key)
+        celebrations = {
+            'should_celebrate_streak': should_celebrate_streak
+        }
```

Comment:
Nit: I thought this would return the number of the streak too. That way the frontend could just accept a number and show it in the celebration as appropriate instead of needing to hardcode or figure that out on its own (Note, this was written before glancing at frontend)